src/main/java/hudson/plugins/locale/PluginImpl.java

```diff
@@ -31,6 +37,14 @@ public class PluginImpl extends GlobalConfiguration {
     private String systemLocale;
     private boolean ignoreAcceptLanguage;
 
+    public static final String USE_BROWSER_LOCALE = "USE_BROWSER_LOCALE";
+
+    // Set of allowed locales
+    private static final Set<String> ALLOWED_LOCALES = new HashSet<>(Arrays.asList(
```

Comment:
Did you test migration from previous version ? What happen if we update the plugin and the local text field is not set ? This plugin already have LocalData migration test. I would like also to have a test that validate this behavior

Reply:
I've added the test case `dataMigration_UnsetLocale` to verify that when the `systemLocale` is not explicitly set and defaults to the browser's locale, the `ignoreAcceptLanguage` flag remains true. This ensures that the plugin continues to enforce the browser's default language setting for all users.

I have tested the migration from previous version. The new additions does not affect the existing tests.